### PR TITLE
Jv rox 18829 switch gke ebpf race condition qa e2e tests to core bpf

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
Switches the race conditions tests to use the core_bpf collection method. Previously an error in stackrox/stackrox/deploy/common/deploy.sh meant that this was not recognized as a valid collection method. See the fix for this at https://github.com/stackrox/stackrox/pull/7436

## Test results

master-gke-race-condition-qa-e2e-tests failed, but only 3 out of 733 tests failed. Collector deployed correctly. The failure was the following 
```
3/733 Tests Failed.
expand_less
SuspiciousLog: Suspicious entries in log file(s) expand_less	0s
{  INFO:  Allow list file: /go/src/github.com/stackrox/stackrox/scripts/ci/logcheck/allowlist-patterns
5945:/tmp/k8s-service-logs/stackrox/pods/central-77fbc4668d-zg6ds-central.log:WARNING: DATA RACE
6640:/tmp/k8s-service-logs/stackrox/pods/central-77fbc4668d-zg6ds-central.log:WARNING: DATA RACE
6778:/tmp/k8s-service-logs/stackrox/pods/central-77fbc4668d-zg6ds-central.log:WARNING: DATA RACE
6918:/tmp/k8s-service-logs/stackrox/pods/central-77fbc4668d-zg6ds-central.log:WARNING: DATA RACE}
```
This doesn't look like it has anything to do with the collection method.

release-x.y-gke-race-condition-qa-e2e-tests failed with the following 
```

SuspiciousLog: Suspicious entries in log file(s) expand_less	0s
{  INFO:  Allow list file: /go/src/github.com/stackrox/stackrox/scripts/ci/logcheck/allowlist-patterns
5936:/tmp/k8s-service-logs/stackrox/pods/central-687b5d6bb6-p89rk-central.log:WARNING: DATA RACE
6590:/tmp/k8s-service-logs/stackrox/pods/central-687b5d6bb6-p89rk-central.log:WARNING: DATA RACE}
ImageManagementTest: Verify risk is properly being attributed to run images expand_more	9m27s
: Run multi-stage test gke-race-condition-qa-e2e-tests - gke-race-condition-qa-e2e-tests-stackrox-e2e container test expand_more	2h9m16s
: Run multi-stage test test phase expand_more
```
It doesn't look like this has anything to do with the collection method.

### CODE REVIEW

**Comment by JoukoVirtanen** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-release-x.y-gke-race-condition-qa-e2e-tests

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-x.y-presubmits.yaml [hunk 1]
```diff
@@ -354,9 +354,9 @@ presubmits:
       - name: result-aggregator
         secret:
           secretName: result-aggregator
-    trigger: (?m)^/test( | .* )gke-qa-e2e-tests,?($|\s.*)
+    trigger: (?m)^/test( | .* )gke-race-condition-qa-e2e-tests,?($|\s.*)
   - agent: kubernetes
-    always_run: false
+    always_run: true
```
Comment (gavin-stackrox) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-x.y-presubmits.yaml:359]: looks like this snuck in

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]

**Review by gavin-stackrox — APPROVED** [review]
/lgtm

**Pull request merged**